homeassistant/components/file/sensor.py

```diff
@@ -42,26 +41,41 @@ async def async_setup_platform(
     config: ConfigType,
     async_add_entities: AddEntitiesCallback,
     discovery_info: DiscoveryInfoType | None = None,
+) -> None:
+    """Set up the file sensor from YAML.
+
+    The YAML platform config is automatically
+    imported to a config entry, this method can be removed
+    when YAML support is removed.
+    """
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
 ) -> None:
     """Set up the file sensor."""
+    config = dict(entry.data)
     file_path: str = config[CONF_FILE_PATH]
-    name: str = config[CONF_NAME]
+    name: str = config.get(CONF_NAME, DEFAULT_NAME)
```

Comment:
name has a default so it will always be set

Reply:
We should keep the name to allow support multiple sensors on the same file as supported in the current code base.

Reply:
The entry data does not have a default name, so we need to add a default incase the name is not set.

Reply:
Right, but we could set it in the schema in the config flow. Perhaps also more obvious to the user?

Reply:
May be, tests need an update probably